### PR TITLE
SCB-373 change Rest args code/encoding to HttpClientFilter/HttpServer…

### PR DESCRIPTION
**Pull request by jeho0815**
…Filter，so some body can do something through writing other Filters around args coding

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: common/common-rest/src/main/java/org/apache/servicecomb/common/rest/filter/inner/ClientRestArgsFilter.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  @Override
+  public void beforeSendRequest(Invocation invocation, HttpServletRequestEx requestEx) {
+    RestClientRequestImpl restClientRequest = (RestClientRequestImpl) invocation.getHandlerContext()
+        .get(RestConst.INVOCATION_HANDLER_REQUESTCLIENT);
+    OperationMeta operationMeta = invocation.getOperationMeta();
+    RestOperationMeta swaggerRestOperation = operationMeta.getExtData(RestConst.SWAGGER_REST_OPERATION);
+    try {
+      RestCodec.argsToRest(invocation.getArgs(), swaggerRestOperation, restClientRequest);
+      requestEx.setBodyBuffer(restClientRequest.getBodyBuffer());
+    } catch (Exception e) {
```
Comment (wujimin) [inline common/common-rest/src/main/java/org/apache/servicecomb/common/rest/filter/inner/ClientRestArgsFilter.java]: it's better to catch throwable

File: common/common-rest/src/main/java/org/apache/servicecomb/common/rest/filter/inner/ServerRestArgsFilter.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    Object body = response.getResult();
+    if (response.isFailed()) {
+      body = ((InvocationException) body).getErrorData();
+    }
+
+    try (BufferOutputStream output = new BufferOutputStream(Unpooled.compositeBuffer())) {
+      produceProcessor.encodeResponse(output, body);
+
+      responseEx.setBodyBuffer(output.getBuffer());
+    } catch (Exception e) {
```
Comment (wujimin) [inline common/common-rest/src/main/java/org/apache/servicecomb/common/rest/filter/inner/ServerRestArgsFilter.java]: it's better to catch throwable

Comment (jeho0815) [reply common/common-rest/src/main/java/org/apache/servicecomb/common/rest/filter/inner/ServerRestArgsFilter.java]: yes, i'll repair this

**Comment by zhengyangyong** [comment]
If user direct invoke rest producer by explorer, explorer will try to get favicon first,then producer will throw an null point exception

**Pull request merged**